testlapack/general.go

```diff
@@ -594,3 +594,55 @@ func isOrthonormal(q blas64.General) bool {
 	}
 	return true
 }
+
+// copyMatrix copies an m×n matrix src of stride n into an m×n matrix dst of stride ld.
+func copyMatrix(m, n int, dst []float64, ld int, src []float64) {
+	for i := 0; i < m; i++ {
+		copy(dst[i*ld:i*ld+n], src[i*n:i*n+n])
+	}
+}
+
+// equalApprox returns whether the matrices A and B of order n are approximately
```

Comment:
In general, we want to test not just that the matrix values are correct, but also that the off-matrix values haven't been modified either. We have found bugs in routines that didn't have these tests, for example a range statement that overwrites all values to zero, not just the in-matrix values. This is why I suggest having slices with random numbers. This fills the whole matrix with meaningful values and checks that only the correct ones have been modified.